gemtext.lisp

```diff
@@ -1,14 +1,24 @@
 (in-package #:phos/gemtext)
 
 (defclass element ()
-  ((text :initarg :text
-         :initform "")))
+  ((text :initform ""
+         :initarg :text
+         :accessor text
+         :type string)))
 
 (defclass title (element)
-  ((level :initarg :level)))
+  ((level :initarg :level
+          :accessor level
+          :type integer
+          :documentation "The nesting level of the title.
+
+Synonymous to the HTML heading levels, i.e. level 1 is <h1> tag, level 2 is <h2> tag etc.")))
```

Comment:
Just a question, there's a way to assert that the level can only be 1, 2 or 3?  (the code handles also other levels, but text/gemini specifies only three levels for the titles.)

Not an issue, yours is already a big improvement on the status quo :)

Reply:
Yes, that is possible with `(integer 1 2)` or `(member 1 2 3)` (I'd prefer the former, as it should be much faster and compiler-friendlier).

I didn't add it because I'm not sure whether all implementations allow such a granular typing. I remember I had issues with CCL when typing slots as `(float 0.0 5.0)`, for example.

Reply:
Agreed, and thanks for the explanation :)